tracer/src/Datadog.Trace/Activity/Handlers/DefaultActivityHandler.cs

```diff
@@ -134,15 +166,25 @@ public void ActivityStopped<T>(string sourceName, T activity)
                         return;
                     }
 
-                    if (ActivityMappingById.TryRemove(activity.Id, out ActivityMapping value) && value.Scope?.Span is not null)
+                    string key;
+                    if (activity is not IW3CActivity w3cActivity)
```

Comment:
super small nit (no necessary to change): I would reverse the condition and do `if (activity is IW3CActivity w3cActivity)`, unless that creates some compilation issue. In my mind that reads easier